gateway/handlers/infohandler.go

```diff
@@ -0,0 +1,62 @@
+package handlers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"io/ioutil"
+	"net/http/httptest"
+
+	"github.com/openfaas/faas/gateway/types"
+	"github.com/openfaas/faas/gateway/version"
+)
+
+// MakeInfoHandler is responsible for display component version information
+func MakeInfoHandler(h http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		responseRecorder := httptest.NewRecorder()
+		h.ServeHTTP(responseRecorder, r)
+		upstreamCall := responseRecorder.Result()
+
+		defer upstreamCall.Body.Close()
+
+		provider := make(map[string]interface{})
+		providerVersion := &types.VersionInfo{}
+
+		upstreamBody, _ := ioutil.ReadAll(upstreamCall.Body)
+		err := json.Unmarshal(upstreamBody, &provider)
+		if err != nil {
+			log.Printf("Error unmarshalling provider json from body %s. Error %s\n", upstreamBody, err.Error())
+		}
+
+		versionMap := provider["version"].(map[string]interface{})
```

Comment:
Should we continue if provider failed to parse? Can we try this out and show what happens with a "fake" endpoint or unit test?